wireapp/wire-ios-mono
# fix: import backup flow - WPB-5885

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5885" title="WPB-5885" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />WPB-5885</a>  [iOS] [timebox] Checking 3.111.3 related issues
  </td></table>
  <br />
 
















----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

When user restores a backup it sometimes end up with an error:
```
 Failed restoring backup: readError(Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory" UserInfo={_kCFStreamErrorCodeKey=2, _kCFStreamErrorDomainKey=1})
```

### Causes (Optional)

Backup file is copied in tmp directory which is cleared when application is resigned or terminated. Tmp directory is used for backup and migration of each steps of database update among other things

### Solutions

- Move clear the temp directory on `applicationDidFinishLaunching`
- Display underlying error in popup error
- Start background activity around db migration to avoid system to kill the app or to allow more time to finish

#### How to Test

On previous app
* Restore backup
* for password prompt go to background or tap passwords to open passwords manager
-> it would fail with error

Now:
Same steps but it should restore the backup without error
